pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -776,6 +799,21 @@ func (r *reconciler) update(lastRealized *lastRealized, newRule *CompletedRule,
 			} else {
 				addedTo := ofPortsToOFAddresses(newOFPorts.Difference(lastRealized.podOFPorts[svcKey]))
 				deletedTo := ofPortsToOFAddresses(lastRealized.podOFPorts[svcKey].Difference(newOFPorts))
+				if isRuleAppliedToService {
```

Comment:
Could a rule be updated from applyingToService to applyingToPod? If yes, I think there is one case need to process; if not, the above two lines should be under `else` condition?

Reply:
It couldn't. `rule.ID` will be different. Changed.